pkg/sysregistries/system_registries.go

```diff
@@ -0,0 +1,98 @@
+package sysregistries
+
+import (
+	"github.com/BurntSushi/toml"
+	"io/ioutil"
+)
+
+/*
+systemRegistriesConfPath is the path to the system-wide registry configuration file
+and is used to add/subtract potential registries for obtaining images.
+You can override this at build time with
+-ldflags '-X github.com/containers/image/sysregistries.systemRegistriesConfPath=$your_path'
+*/
+var systemRegistriesConfPath = builtinRegistriesConfPath
+
+/*
+builtinRegistriesConfPath is the path to registry configuration file
+DO NOT change this, instead see systemRegistriesConfPath above.
+*/
+const builtinRegistriesConfPath = "/etc/containers/registries.new"
+
+type tomlConfig struct {
+	Registries registries `toml:"registries"`
+}
+
+type registries struct {
+	SecureRegistries   []string `toml:"secure"`
+	InsecureRegistries []string `toml:"insecure"`
+	BlockRegistries    []string `toml:"block"`
+}
+
+var blankFile = []byte("")
+
+/*
+Reads the global registry file from the filesystem. Returns
+a byte array
+*/
+func readRegistryConf() ([]byte, error) {
+	configBytes, err := ioutil.ReadFile(systemRegistriesConfPath)
+	if err != nil {
+		return blankFile, err
+	}
+	return configBytes, nil
+}
+
+// For mocking in unittests
+var readConf = readRegistryConf
+
+/*
+Loads the registry configuration file from the filesystem and
+then unmarshals it.  Returns the unmarshalled object.
+*/
+func loadRegistryConf() (*tomlConfig, error) {
+	var config tomlConfig
```

Comment:
also, initialize this as `config := &tomlConfig{}` so this is already a pointer and  you don't have to use `&`